saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequestFactory.java

```diff
@@ -27,15 +33,44 @@
  */
 public interface Saml2AuthenticationRequestFactory {
 	/**
-	 * Creates an authentication request from the Service Provider, sp,
-	 * to the Identity Provider, idp.
+	 * Creates an authentication request from the Service Provider, sp, to the Identity Provider, idp.
 	 * The authentication result is an XML string that may be signed, encrypted, both or neither.
+	 * This method only returns the {@code SAMLRequest} string for the request, and for a complete
+	 * set of data parameters please use {@link #createAuthenticationRequestParameters(Saml2AuthenticationRequest)}
 	 *
 	 * @param request - information about the identity provider, the recipient of this authentication request and
 	 *                accompanying data
 	 * @return XML data in the format of a String. This data may be signed, encrypted, both signed and encrypted or
 	 * neither signed and encrypted
 	 * @throws Saml2Exception when a SAML library exception occurs
+	 * @deprecated Use {@link #createAuthenticationRequestParameters(Saml2AuthenticationRequest)} to retrieve all parameters
```

Comment:
Can you clarify why this is deprecated now? IOW, why would it be wrong or undesireable for an app to invoke or implement this method?

Reply:
It's definition results in an invalid, albeit working for many major if not most IDP implementations, message encoding.